modules/caddyhttp/reverseproxy/httptransport.go

```diff
@@ -141,6 +143,22 @@ type HTTPTransport struct {
 	// The pre-configured underlying HTTP transport.
 	Transport *http.Transport `json:"-"`
 
+	// The module that provides the network (forward) proxy
+	// URL that the HTTP transport will use to proxy
+	// requests to the upstream. See [http.Transport.Proxy](https://pkg.go.dev/net/http#Transport.Proxy)
+	// for information regarding supported protocols.
+	//
+	// Providing a value to this parameter results in requests
+	// flowing through the reverse_proxy in the following way:
+	//
+	// User Agent ->
+	//  reverse_proxy ->
+	//  [proxy provided by the module] -> upstream
+	//
+	// If nil, defaults to reading the `HTTP_PROXY`,
+	// `HTTPS_PROXY`, and `NO_PROXY` environment variables.
+	NetworkProxyRaw json.RawMessage `json:"network_proxy,omitempty" caddy:"namespace=caddy.network_proxy.source inline_key=from"`
```

Comment:
Are there other modules that could be made under `caddy.network_proxy`? (i.e. is the `.source` namespace useful)?

If so, let's definitely keep it -- just curious.

Reply:
I can't think of anything, but I put it there just in case. If you feel it's unnecessary and there are no possible other modules, I can remove it.

Reply:
I don't know what else might go there... I suppose `caddy.network_proxy` might be a sufficient namespace.

Reply:
Changed!